hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -157,14 +157,22 @@ public void accept(MessageSendingResult result) {
                 rateLimiter.registerSuccessfulSending();
                 handleMessageSendingSuccess(message, result);
             } else {
-                handleFailedSending(message, result);
-                message.incrementRetryCounter();
+                if (result.isUnauthorized()) {
+                    message.incrementNumberOfUnauthorizedRequests();
+                }
```

Comment:
This code is really weird... What is your approach for handling unauthorized requests? Should we retry or not? Should we discard messages? Should we stop and reduce outgoing rate?